docs/ref/models/indexes.txt

```diff
@@ -30,16 +30,66 @@ options`_.
 
 .. attribute:: Index.fields
 
-A list or tuple of the name of the fields on which the index is desired.
+A list or tuple of the name of the fields or database expressions on which the index is desired.
 
 By default, indexes are created with an ascending order for each column. To
 define an index with a descending order for a column, add a hyphen before the
-field's name.
+field's name. When using an expression, use
+:meth:`~django.db.models.Expression.asc()` for ascending (default) and
+:meth:`~django.db.models.Expression.desc()` for descending order.
 
 For example ``Index(fields=['headline', '-pub_date'])`` would create SQL with
 ``(headline, pub_date DESC)``. Index ordering isn't supported on MySQL. In that
 case, a descending index is created as a normal index.
 
+``Index(fields=[Lower('title').desc()], name='lower_title_idx')`` would create
+an index on the lowercase value of the ``title`` field in descending order.
+Using ``F`` expressions in a construct like
+``Index(fields=[F('height') * F('weight')], name='math_idx')`` allows
+constructing indexes on mathematical operations.
+
+.. admonition:: Database specific exceptions of suitable expressions
```

Comment:
https://mariadb.com/kb/en/library/functional-indexes/ Could you please mention this? MariaDB is officially supported by Django.